Develop

yapracticumplakhotnik opened the pull request:
Финальный проект 6 спринта
Тестовые сценарии
1)Выпадающий список в разделе «Вопросы о важном». 
2)Заказ самоката.

Commit:
test

Commit:
test

pomegranate282922 commented on src/main/java/pages/FAQPage.java:
```diff
@@ -0,0 +1,48 @@
+    //восьмой вопрос
+    public By question8 = By.xpath(".//div[@class='accordion__item']//div[@id=\"accordion__heading-7\"]");
+    //восьмой ответ
+    public By actual8 = By.xpath("/html/body/div/div/div/div[5]/div[2]/div/div[8]/div[2]/p");
```
Используем относительные локаторы. Абсолютные слишком нестабильны. Нужно использовать не более 4х переходов.
Нужно исправить

pomegranate282922 commented on src/test/java/tests/FAQTest.java:
```diff
@@ -0,0 +1,152 @@
+        driver = new FirefoxDriver();
+
+        faqPage = new FAQPage(driver);
+        driver.get("https://qa-scooter.praktikum-services.ru/");
```
Урлы лучше вынести как константы 
Можно улучшить

pomegranate282922 commented on src/test/java/tests/FAQTest.java:
```diff
@@ -0,0 +1,152 @@
+        //Прокрутка
+        WebElement element = driver.findElement(faqPage.question1);
+        ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView(true);", element);
+        Thread.sleep(500);
```
Все слипы нужно заменить на явные ожидания 
Нужно исправить

pomegranate282922 commented on src/test/java/tests/FAQTest.java:
```diff
@@ -0,0 +1,152 @@
+        String expected = "Да, обязательно. Всем самокатов! И Москве, и Московской области.";
+        String actual = driver.findElement(faqPage.actual8).getText();
+        assertEquals(expected, actual, "Ошибка текст не равен");
+    }
```
Здесь ждем применения параметризации 
Нужно исправить

pomegranate282922 commented on src/test/java/tests/FAQTest.java:
```diff
@@ -0,0 +1,152 @@
+
+    @AfterEach
+    public void tearDown() {
+        if (driver != null) {
```
Общие начала и концы всех тестовых классов удобно вынести в общий класс предок
Можно улучшить

pomegranate282922 commented on src/test/java/tests/OrderTest.java:
```diff
@@ -0,0 +1,54 @@
+    @CsvSource({
+            "top, Иван, Иванов, Москва, Черкизовская, +79991234567, 10.10.2023, сутки, black, Позвоните за 10 минут",
+            "bottom, Петр, Петров, Москва, Сокольники, +79997654321, 15.10.2023, двое суток, grey, Позвоните за 5 минут"
+    })
```
Отлично применена параметризация

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Применена параметризация.
Замечания:
1. Применить параметризацию в тестах на faq 
2. Использовать более стабильные локаторы 
3. Убираем слипы 
4. Коммитим только папку src и пом файл, если в задании явно не указано иное

Commit:
правки:

1) исправлены локаторы ответов
2) урлы вынесены в Constant
3) убраны слипы
4) применена параметризация для класса FAQTest

pomegranate282922 commented on src/test/java/tests/FAQTest.java:
```diff
@@ -0,0 +1,70 @@
+            "question8, actual8, 'Да, обязательно. Всем самокатов! И Москве, и Московской области.'"
+    })
+    public void testFaqAccordionItem(String questionLocatorName, String actualLocatorName, String expectedText) throws InterruptedException {
+        By questionLocator = faqPage.getLocatorByName(questionLocatorName);
```
Рефлексия это здорово, но лучше бы проще) можно передавать номер и уже по номеру в page object составлять нужный локатор 
Можно улучшить

pomegranate282922 reviewed (approved):
Ок, зачет!
